eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -297,6 +299,34 @@ object LightningMessageCodecs {
 
   //
 
+  val knownUnknownMessageCodec: Codec[KnownUnknownMessage] = (
+    ("tag" | uint16) ::
+      ("data" | varsizebinarydata)
+    ).as[KnownUnknownMessage]
+
+  val unknownUnknownMessageCodec: Codec[UnknownUnknownMessage] = (
+    ("tag" | uint16) ::
+      ("message" | knownUnknownMessageCodec)
+    ).as[UnknownUnknownMessage]
+
```

Comment:
I don't understand why you want to distinguish between these two.
Why can't you just have a single `UnknownMessage` that you use in the fallback of the discriminator?

Reply:
indeed